nemo/collections/nlp/models/language_modeling/megatron_base_model.py

```diff
@@ -527,7 +533,7 @@ def configure_optimizers(self):
             assert self._trainer.max_steps is not None, "'max_steps' is missing in trainer config."
             if hasattr(self._cfg.optim, 'sched'):
                 sched_config = self._cfg.optim.sched
-                sched_config['max_steps'] = self._trainer.max_steps
+                assert sched_config.get('max_steps', -1) >= 0
```

Comment:
Why should this be set manually on the config above ? Trainer has this value

Reply:
The whole point of this PR is to fix the situation where `trainer.max_steps` is `None`. We need `cfg.optim.sched.max_steps` to be set for the scheduler to work, so we use `trainer.max_epochs` to compute `max_steps` in this situation (in the new `_get_max_steps()` function added in this PR).

This assert is a sanity check, it shouldn't possible for it to fail in the current code since we do `optim_config.sched.max_steps = self._get_max_steps()`. I added it to (i) make sure my code is working as intended, and (ii) make sure no-one breaks it in the future.

Reply:
> oh, I just noticed the assert self._trainer.max_steps is not None above... now I'm a bit confused as to how I was testing this, it's been a long time and I don't remember :/

After testing it I now remember: it's because `max_steps` is set to -1 when undefined (which is confusing because `max_epochs` is using `None`). I think this assert was actually a mistake, assuming that a `None` value could appear when undefined, but I tested setting `max_steps: null` in the config and it actually crashes earlier during PTL trainer initialization because `None` is not valid.

=> I removed this assert in ff09438c5 (and added another one that validates that nobody uses a negative value for `max_steps` in the scheduler config, since we use `None` here for undefined).